docs/tutorial/typesfuns.rst

```diff
@@ -1182,3 +1183,71 @@ matching ``let`` and lambda bindings. It will *only* work if:
 - The type of the result is "known". i.e. the type of the expression
   can be determined *without* type checking the ``case``-expression
   itself.
+
+Totality
+========
+
+A *total* function is a function that terminates for all possible
+inputs, or one that is guaranteed to produce some output before making
+a recursive call. For example, Idris' ``head`` function is total for
+all lists:
+
+::
+
+    Idris> :t Prelude.List.head
+    head : (l : List a) -> {auto ok : NonEmpty l} -> a
+
+The ``{auto ok : NonEmpty l}`` tells us that Idris won't compile if we
+try to call ``head`` on an empty list. (Note that this is in contrast to
+Haskell's ``head``, which is  *not* total and will fail at runtime rather
```

Comment:
Maybe it is a good idea to move the unsafe version of head to here, and then show both versions in action...